packages/next/build/index.ts

```diff
@@ -238,21 +238,40 @@ export default async function build(
         }`,
       })
 
-      const verifyResult = await nextBuildSpan
-        .traceChild('verify-typescript-setup')
-        .traceAsyncFn(() =>
-          verifyTypeScriptSetup(
+      const ignoreESLint = Boolean(config.eslint.ignoreDuringBuilds)
+      const eslintCacheDir = path.join(cacheDir, 'eslint/')
+      const shouldLint = !ignoreESLint && runLint
+
+      const [[verifyResult, typeCheckEnd]] = await Promise.all([
+        nextBuildSpan.traceChild('verify-typescript-setup').traceAsyncFn(() =>
+          validiteTypeScriptTypes(
             dir,
             [pagesDir, viewsDir].filter(Boolean) as string[],
             !ignoreTypeScriptErrors,
             config,
-            cacheDir
-          )
-        )
-
-      const typeCheckEnd = process.hrtime(typeCheckStart)
+            cacheDir,
+            config.experimental.cpus,
+            config.experimental.workerThreads
+          ).then((resolved) => {
+            const checkEnd = process.hrtime(typeCheckStart)
+            typeCheckingSpinner?.stopAndPersist()
```

Comment:
Since we are grouping these together now should we update the spinner to mention when both are running and wait to stop it until both are finished?

Reply:
Sounds a good idea, but the status message might be a little complex:

```js
      let typeCheckingAndLintingSpinnerPrefixText = Log.prefixes.info + ' ';
      if (ignoreTypeScriptErrors && shouldLint) {
        typeCheckingAndLintingSpinnerPrefixText += 'Linting and skipping validation of types'
      } else if (!ignoreTypeScriptErrors && shouldLint) {
        typeCheckingAndLintingSpinnerPrefixText += 'Linting and checking validity of types'
      } else if (ignoreTypeScriptErrors && !shouldLint) {
        typeCheckingAndLintingSpinnerPrefixText += 'Skipping lint and validation of types'
      } else {
        typeCheckingAndLintingSpinnerPrefixText += 'Skipping lint and checking validity of types'
      }
```

Any ideas about the message?

Reply:
The message could be something like:

```js
if (!ignoreTypeScript && shouldLint) {
  text = 'Linting and checking validity of types'
} else if (!ignoreTypeScript) {
  text = 'Checking validity of types'
} else if (shouldLint) {
  text = 'Linting'
}
```
And then we could log the `Skipped linting` or `Skipped checking validation of types` separately maybe